src/utilities/api-requests.ts

```diff
@@ -10,13 +15,23 @@ import axios from "axios";
  */
 export async function request<T>(uri: string, defaultValue: T): Promise<T> {
     let data = defaultValue;
-    await axios
-        .get(uri)
-        .then((res) => {
-            data = res.data as T;
-        })
-        .catch((error) => {
-            console.error(`Request to ${uri} failed with\n${error}`);
-        });
+    let attempt = 0;
+
+    while (data === defaultValue && attempt < 5) {
+        const response = await axios
+            .get<T>(uri)
+            .catch((reason: AxiosError) => handleError(uri, reason));
+
+        if (response) {
+            if (response.status === 200) {
+                data = response.data;
+            } else if (response.status === 429) {
+                await new Promise((resolve) => setTimeout(resolve, 30000));
```

Comment:
Magic number (though it is reasonably clear what it is, it wouldn't hurt to have some variable named with what 30000 represents).

Reply:
Resolved.